### ISSUE TITLE
[Feature request] ReportUtilityBills option to also calculate by End Use

### ISSUE THREAD

**Issue by jmarrec** [NREL/OpenStudio-HPXML#1352]
`ReportSimulationOutput` has the capability, when `include_timeseries_end_use_consumptions` is true, to output hourly timeseries by Fuel and End Use.

It would be great to be able to leverage that in the `ReportUtilityBills` measure, so one could see, for each fuel type,  how much each end use contributes to the total, especially when a non flat structure is used (for a flat structure you can get the annual value by end use and multiply by the flat rate, but that's not true for TOU or tiered rates).

**Comment by shorowit** [comment]
I suppose this could be allowed for flat and TOU rates, but I don't see how it would work for tiered rates. If your tiered structure is 10 cents/kWh up to 500 kWh for the month, and then 15 cents/kWh after, and you have End Use A that consumes 400 kWh and End Use B that consumes 200 kWh for the month, how would you attribute dollars to each? Suppose you simply apportioned them by their consumption like so:
Total $ = 500 * 0.1 + 100 * 0.15 = $65
End Use A $ = $65 * 400/600 = $43.33
End Use B $ = $65 * 200/600 = $21.67
That may seem fine, but then if you applied an upgrade that completely eliminated End Use B consumption, your utility bills would be reduced by $25, not $21.67. It seems confusing to me.

That said, it does make me think that we should at least offer monthly utility bill outputs by fuel type. Not sure why we don't do that.

**Comment by shorowit** [comment]
Also tiered rates can _only_ be calculated monthly. Any other rates can be calculated hourly, I think.

**Comment by jmarrec** [comment]
Yeah for tiered I think you'd apportion the monthly values per end use, then roll up to get the annual $/end use.
Getting an annual breakdown of cost per end use would be nice enough, monthly would be great. The hourly stuff is too granular to be meaningful anyways I think
